index.js

```diff
@@ -1687,7 +1742,58 @@
   //. //
   //. //   The value at position 1 is not a member of ‘{ x :: FiniteNumber, y :: FiniteNumber }’.
   //. ```
-  function RecordType(fields) {
+  //.
+  //. ##### Extending record types
+  //.
+  //. By using the `parent` argument of the RecordType constructor, one can
+  //. define record types which must contain all the same properties of
+  //. another. For example, one could define `Poinst3D` as a refinement of
+  //. `Point` from the previous example:
+  //.
+  //. ```js
+  //. //    Point3D :: Type
+  //. const Point3D = $.RecordType(Point, {z: $.FiniteNumber});
+  //.
+  //. //    a :: Point
+  //. const a = {x: 0, y: 0};
+  //.
+  //. //    b :: Point, Point3D
+  //. const b = {x: 3, y: 4, z: 5};
+  //.
+  //. Point.validate(a)   // => Right(a)
+  //. Point3D.validate(a) // => Left({propPath: ['z'], value: undefined})
+  //.
+  //. Point.validate(b)   // => Right(b)
+  //. Point3D.validate(b) // => Right(b)
+  //.
+  //. dist(a, b);
+  //. // => 5
+  //. ```
+  //.
+  //. The `b` value is a member of `Point3D` as well as `Point`, allowing one
+  //. to call `dist` on it. By default record types allow additional
+  //. properties, permitting record "extension". If one wishes for a record
+  //. type to allow for no additional properties, it can be refined as such:
+  //.
+  //. ```js
+  //. //    strict :: Type -> Type
+  //. const strict = t => $.NullaryType(
```

Comment:
Should this be named `Strict`, to match [`$.NonEmpty`][1]? Similarly, could it be a unary type so one would define `Strict Point` rather than `StrictPoint`?


[1]: https://github.com/sanctuary-js/sanctuary-def#NonEmpty